pkg/router/controller/factory/factory.go

```diff
@@ -326,3 +384,38 @@ func (r routeAge) Swap(i, j int) { r[i], r[j] = r[j], r[i] }
 func (r routeAge) Less(i, j int) bool {
 	return routeapihelpers.RouteLessThan(&r[i], &r[j])
 }
+
+func endpointSliceServiceName(eps *discoveryv1beta1.EndpointSlice) string {
+	if name, ok := eps.Labels[ServiceNameLabel]; ok && name != "" {
+		return name
+	}
+	return ""
+}
+
+func endpointSliceByServiceLabelIndexFunc(obj interface{}) ([]string, error) {
+	if eps, ok := obj.(*discoveryv1beta1.EndpointSlice); ok {
```

Comment:
This function will always return a `nil` error.

Reply:
I think that's fine. If there's no label then there's nothing to index.